### PR TITLE
Add code activities for ch7 in ep3 (broken ship)

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Resolve `07-broken-ship` in #1

### CODE REVIEW

File: knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    public function grade(CodingExecutionResult $result)
+    {
+        $deathStar3Class = new \ReflectionClass('\DeathStarIII');
+        if (!$deathStar3Class->isSubclassOf('\AbstractDeathStar')) {
+            throw new GradingException('The `DeathStarIII` class should extend the `AbstractDeathStar` one.');
+        }
+        $deathStar3 = $result->getDeclaredVariableValue('deathStar3');
+        if (!$result->doesOutputContain($deathStar3->getDescription())) {
+            throw new GradingException('Hmm, did you print the `DeathStarIII` description in `h3` tag?');
+        }
```
Comment (weaverryan) [inline knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php:97]: nice way to grade this

File: knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        \$range = \$this->getLaserRange();
+
+        return <<<HEREDOC
+A fantastic death machine, made to be extra cold and
+intimidating. It comes complete with a "super-laser"
+capable of destroying planets, with a range of \$range.
+HEREDOC;
+    }
+}
+EOF
```
Comment (weaverryan) [inline knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php]: This class didn't change, right? So it doesn't need to be listed down here.

**Comment by weaverryan** [comment]
One one comment - looks great!

File: knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php (hunk 1 above)
Comment (bocharsky-bw) [inline knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php:97]: @weaverryan Do we need to check that description really in `h3` tag? Or leave as is?

File: knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php (hunk 2 above)
Comment (bocharsky-bw) [inline knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php]: Yes, I dropped it.

File: knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php (hunk 1 above)
Comment (weaverryan) [inline knpu/episode3/Challenges/BrokenShip/CreateDeathStarIIICoding.php:97]: I liked that you didn't check it - we don't really care about the placement, just that the user proves they made the changes to the classes that would result in the correct description being printed somewhere.

**Pull request merged**